vessel-runtime/src/main/java/com/textnow/android/vessel/VesselImpl.kt

```diff
@@ -185,32 +197,28 @@ class VesselImpl(
                 preloadImpl(timeoutMS)
             }
         }
-
-        if (!loadCompleted) {
-            profiler.count(Event.PRELOAD_TIMEOUT)
-        }
     }
 
-    override fun preloadBlocking(timeoutMS: Int?) {
+    override fun preloadBlocking(timeoutMS: Int?): PreloadReport {
         if (cache == null) {
-            return
+            return PreloadReport(errorsOcurred = true)
         }
 
-        val loadCompleted = profiler.timeBlocking(Span.PRELOAD_FROM_DB) {
+        return profiler.timeBlocking(Span.PRELOAD_FROM_DB) {
             preloadImpl(timeoutMS)
         }
-
-        if (!loadCompleted) {
-            profiler.countBlocking(Event.PRELOAD_TIMEOUT)
-        }
     }
 
     // endregion
 
     // region profiling
 
     override val profileData
-        get() = profiler.snapshot
+        get() = if (DummyProfiler::class.isInstance(profiler)) {
```

Comment:
it doesn't seem like VesselImpl.kt should know about the different types of Profiler that can be passed in.  Is there a reason the DummyProfiler can not return null for the snapshot?

Reply:
I'd prefer the public contract of `Profiler` to disallow `null` for `snapshot` - I don't see a normal case where a profiler would return a null snapshot, since that's really it's only job.

I added `DummyProfiler` as a way to avoid branching all the VesselImpl code that uses `time()` or `timeBlocking()` when profiling is disabled.

Since the caller needs a way to know whether the snapshot is valid or not (profiling is disabled or not) this getter needs to differentiate somehow.

The other option would be to just always enable profiling and not make it a config option.  Then this is always valid.  It was made optional initially due to concerns raised about performance, but I think we can just leave it on now.  Profiling is enabled for most users today anyways.